src/PollingBlockTracker.test.ts

```diff
@@ -0,0 +1,3026 @@
+          await setTimeoutRecorder.nextMatchingDuration(
+            blockTrackerOptions.blockResetDuration,
+          );
+          expect(blockTracker.getCurrentBlock()).toBeNull();
+        },
+      );
+    });
+  });
+
+  describe('checkForLatestBlock', () => {
+    it('should start the block tracker shortly after being called', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(
+        {
+          provider: {
+            stubs: [
+              {
+                methodName: 'eth_blockNumber',
+                response: {
+                  result: '0x0',
+                },
+              },
+              {
+                methodName: 'eth_blockNumber',
+                response: {
+                  result: '0x1',
+                },
+              },
+            ],
+          },
+        },
+        async ({ blockTracker }) => {
+          blockTracker.checkForLatestBlock();
+          await new Promise((resolve) => {
+            blockTracker.on('latest', resolve);
+          });
+          expect(blockTracker.isRunning()).toBe(true);
+        },
+      );
+    });
+
+    it('should stop the block tracker automatically after its promise is fulfilled', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(async ({ blockTracker }) => {
+        await blockTracker.checkForLatestBlock();
+        expect(blockTracker.isRunning()).toBe(false);
+      });
+    });
+
+    it('should fetch the latest block number', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(
+        {
+          provider: {
+            stubs: [
+              {
+                methodName: 'eth_blockNumber',
+                response: {
+                  result: '0x0',
+                },
+              },
+            ],
+          },
+        },
+        async ({ blockTracker }) => {
+          const latestBlockNumber = await blockTracker.checkForLatestBlock();
+          expect(latestBlockNumber).toStrictEqual('0x0');
+        },
+      );
+    });
+
+    it('request the latest block number with `skipCache: true` if the block tracker was initialized with `setSkipCacheFlag: true`', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(
+        { blockTracker: { setSkipCacheFlag: true } },
+        async ({ provider, blockTracker }) => {
+          jest.spyOn(provider, 'sendAsync');
+
+          await blockTracker.checkForLatestBlock();
+
+          expect(provider.sendAsync).toHaveBeenCalledWith(
+            {
+              jsonrpc: '2.0' as const,
+              id: expect.any(Number),
+              method: 'eth_blockNumber' as const,
+              params: [],
+              skipCache: true,
+            },
+            expect.any(Function),
+          );
+        },
+      );
+    });
+
+    METHODS_TO_ADD_LISTENER.forEach((methodToAddListener) => {
```

Comment:
It looks like `methodToAddListener` isn't being used in these tests, so this loop seems unnecessary

Reply:
Ah, looks like some copypasta. Removed in [ec4a03e](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/ec4a03e3fe88c05e35794b2dc8f1f5dc98bb13f4)